src/frontend/src/optimizer/plan_node/logical_over_agg.rs

```diff
@@ -282,6 +308,19 @@ impl PlanTreeNodeUnary for LogicalOverAgg {
     fn clone_with_input(&self, input: PlanRef) -> Self {
         Self::new(self.window_functions.clone(), input)
     }
+
+    #[must_use]
+    fn rewrite_with_input(
+        &self,
+        input: PlanRef,
+        input_col_change: ColIndexMapping,
+    ) -> (Self, ColIndexMapping) {
+        let over_agg =
+            self.rewrite_with_input_window(input, &self.window_functions, input_col_change);
+        // change the input columns index will not change the output column index
```

Comment:
This seems incorrect. Its schema is the input schema plus the window functions, so it would change.

Reply:
BTW, Do we need this function? It seems unused.

Reply:
Oh, this seems correct as `rewrite_with_input_window` doesn't change the input cols... Slightly confusing 😇

Reply:
This function may be used later when implement ToStream trait for this node.

Reply:
c.c. @richardchien PTAL thanks